ebos/eclbasevanguard.hh

```diff
@@ -444,6 +466,45 @@ public:
             parallelWells_.emplace_back(well.name(), true);
         }
         std::sort(parallelWells_.begin(), parallelWells_.end());
+
+        // Check whether allowing distribute wells makes sense
+        if (enableDistributedWells() )
+        {
+            int hasMsWell = false;
+
+            if (EWOMS_GET_PARAM(TypeTag, bool, UseMultisegmentWell))
+            {
+                if (myRank == 0)
+                {
+                    const auto& wells = this->schedule().getWellsatEnd();
+                    for ( const auto& well: wells)
+                    {
+                        hasMsWell = hasMsWell || well.isMultiSegment();
+                    }
+                }
+            }
+#if DUNE_VERSION_NEWER(DUNE_COMMON, 2, 7)
+            const auto& comm = Dune::MPIHelper::getCommunication();
+#else
+            const auto& comm = Dune::MPIHelper::getCollectiveCommunication();
+#endif
+            hasMsWell = comm.max(hasMsWell);
+
+            if (hasMsWell)
+            {
+                if (myRank == 0)
+                {
+                    std::string message = std::string("Option --allow-distributed-wells=true is only allowed ")
+                        + "if model only has only standard wells."
+                        + "You need to provide option "
+                        + " with --enable-multisegement-wells=false to treat "
+                        + "existing multisegment wells as standard wells.";
```

Comment:
This will be a really long line in the log - add a newline somewhere?

Reply:
Good idea. Changed.